java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/ArrowVectorIterator.java

```diff
@@ -32,7 +32,7 @@
 /**
  * VectorSchemaRoot iterator for partially converting JDBC data.
  */
-public class ArrowVectorIterator implements Iterator<VectorSchemaRoot> {
+public class ArrowVectorIterator implements Iterator<VectorSchemaRoot>, AutoCloseable {
```

Comment:
Add `@Override` to `close()`?

Reply:
Nice catch, we should see if we can add ErrorProne or another static analysis tool to catch these things automatically.

Reply:
done.